espnet2/train/trainer.py

```diff
@@ -362,7 +366,12 @@ def run(
                     options=trainer_options,
                     distributed_option=distributed_option,
                 )
-            epoch_stats = dp_model.validation_epoch_end_()
+            if distributed_option.distributed or (distributed_option.ngpu > 1):
+                if distributed_option.dist_rank == 0:
+                    epoch_stats = dp_model.module.validation_epoch_end_()
+            else:
+                epoch_stats = dp_model.validation_epoch_end_()
+
             if epoch_stats is not None:
                 reporter.register_epoch_stats("valid", stats=epoch_stats)
```

Comment:
These two should be inside the if I think, no? Otherwise it runs on all gpus. @juice500ml